.github/ISSUE_TEMPLATE/release.md

```diff
@@ -0,0 +1,43 @@
+---
+name: Release
+about: When ready to cut a release
+title: Release X.Y.Z
+labels: release
+assignees: ''
+
+---
+
+- [ ] Start a new release branch:
+```bash
+$ git flow release start X.Y.Z
+```
+- [ ] Rotate `CHANGELOG.md` (following [Keep a Changelog](https://keepachangelog.com/) principles)
+- [ ] Ensure outstanding changes are committed:
+```bash
+$ git status # Ensure we have a clean staging area
+$ git add CHANGELOG.md
+$ git commit -m "X.Y.Z"
+```
+- [ ] Publish the release branch:
+```bash
+$ git flow release publish X.Y.Z
+```
+- [ ] Test the release branch on staging
```

Comment:
I would consider removing all of the sub-bullets under here.

Reasoning: If we need to explain something in detail, I would prefer to keep it separate from the checklist (deployment `README`, wiki, etc.). The hope is that minimizing what's in the checklist will allow the people running through it to stay focused on executing each step in the process.